python/graphstorm/dataloading/metadata.py

```diff
@@ -0,0 +1,954 @@
+            "graph_type": "heterogeneous",
+            "nodes": [
+                {
+                    "node_type": "ntype1",
+                    "features": [{
+                        "feat_name": "feat1",
+                        "feat_dim": [4, 7]
+                    },
+                                {
+                        "feat_name": "feat1",
+                        "feat_dim": [32]
+                    }
+                    ]
+                },
+                {
+                    "node_type": "ntype2",
+                    "features": []
+                }
+            ],
+            "edges": [
+                {
+                    "source_node_type": "ntype1",
+                    "edge_type": "etype1",
+                    "destination_node_type": "ntype2",
+                    "features": [
+                        {
+                            "feat_name": "feat",
+                            "feat_dim": [3]
+                        }
+                    ]
+                }
+            ]
+        }
+
+        Return
+        -------
+        dict: the hierarchy structure of the graph metadata like the example above.
+        """
+        metadata_dict = {"graph_type": self._gtype}
+        nodes = []
+        for ntype in self._ntypes:
+            node = {"node_type": ntype}
+            if ntype in self._nfeat_dims:
+                nfeats = []
+                for feat_name, feat_dim in self._nfeat_dims[ntype].items():
+                    nfeat_dim = {
+                        "feat_name": feat_name,
+                        "feat_dim": feat_dim
+                        }
+                    nfeats.append(nfeat_dim)
+                node['features'] = nfeats
+            else:
+                node['features'] = []
+
+            nodes.append(node)
+
+        metadata_dict["nodes"] = nodes
+
+        edges = []
+        for src_ntype, etype, dst_ntype in self._etypes:
+            edge = {
+                "source_node_type": src_ntype,
+                "etype": etype,
+                "destination_node_type": dst_ntype
+            }
+            if (src_ntype, etype, dst_ntype) in self._efeat_dims:
+                efeats = []
+                for feat_name, feat_dim in self._efeat_dims[(src_ntype, etype, dst_ntype)].items():
+                    efeat_dim = {
+                        "feat_name": feat_name,
+                        "feat_dim": feat_dim
+                        }
+                    efeats.append(efeat_dim)
+                edge['features'] = efeats
+            else:
+                edge['features'] = []
+
+            edges.append(edge)
+
+        metadata_dict["edges"] = edges
+
+        return metadata_dict
+
+    def to_string(self):
+        """ Convert the graph metadata to a simple dictionary string.
+
+        Return
+        -------
+        str: string of the graph metadata dictionary.
+        """
+        metadata_dict = self.to_dict()
+        return str(metadata_dict)
```

Comment:
What's the intended use of this function? The result will not be valid JSON and is not readable either.

Reply:
This is for print this class only. If users print an instance of this class, the output will be `<GSGraphMetadata.*****>`. So this `to_string()` function overwrite the default one.